Polkadot TPS FAQ

CrackTheCode016 opened the pull request:
Closes #4908 

- Adds important precursor information for TPS in Polkadot (and the metric in general)


## todo

- [x] Need some sort of concrete number of what it _could _ achieve, and how it does it.

CrackTheCode016 committed:
faq

CrackTheCode016 committed:
add details

CrackTheCode016 committed:
grammar

CrackTheCode016 commented:
Note to reviewers: this needs a bit of review to make sure the numbers are concrete and accurate.

filippoweb3 commented on docs/general/faq.md:
```diff
@@ -161,6 +161,21 @@ imposes economic costs on the operators of the bridge, and therefore makes it ec
+[100,000 to 1,000,000 transactions per second.](https://polkadot.network/blog/polkadot-roadmap-roundup).
+This is primarily due to the pending [asynchronous backing upgrade](../learn/learn-async-backing).
+
+It is essential to realize that TPS is inherently a subjective measurement with numerous factors
```
Which factors? It would be nice to mention a couple of them.

filippoweb3 commented on docs/general/faq.md:
```diff
@@ -161,6 +161,21 @@ imposes economic costs on the operators of the bridge, and therefore makes it ec
+that can contribute to it. It's hard to gauge the usefulness of TPS in isolation, as it's highly
+dependent on what a transaction does for a particular network.
+
+Furthermore, there is no industry standard for measuring TPS non-biasedly.
```
```suggestion
```

filippoweb3 replied:
Unless we can back up this statement with some docs / papers I would leave it out as it is a bit strong

CrackTheCode016 replied:
I'll adjust using this resource: https://github.com/paritytech/polkadot-stps

filippoweb3 commented on docs/general/faq.md:
```diff
@@ -161,6 +161,21 @@ imposes economic costs on the operators of the bridge, and therefore makes it ec
+Furthermore, there is no industry standard for measuring TPS non-biasedly.
+
+In the case of Polkadot, the TPS can be very high due to the parallelization of its network. Because
+it is effectively finalizing the state of many parachains at once.
```
```suggestion
In the case of Polkadot, TPS can be high if we consider parallelization. Because Polkadot is sharded network,
it is effectively finalizing the state of many parachains at once.
```

filippoweb3 reviewed (changes requested):
Nice addition, added a couple of edits and a request :)

profradha commented on docs/general/faq.md:
```diff
@@ -161,6 +161,21 @@ imposes economic costs on the operators of the bridge, and therefore makes it ec
 there will be bridges between Polkadot and most of the other major chains.
 
+### What is Polkadot's Transactions Per Second (TPS)?
+
```
```suggestion
Polkadot is a heterogeneous sharded network comprising a relay chain and numerous parachains, which are all individual blockchains built on [Substrate](https://substrate.io/) executing in parallel. Hence, the Transactions Per Second (TPS) of Polkadot is a number that encompasses all the transactions on the relay chain as well as parachains. As the transactions on these Substrate-based blockchains are [weights based](https://docs.substrate.io/build/tx-weights-fees/), it makes sense to use TPS as a measure for the network performance if all the transactions carry the same weight. [Performance benchmark tests](https://twitter.com/gavofyork/status/1270025498580656134) show that Substrate based blockchains can achieve over 1000 TPS for balance transfer transactions. Assuming Polkadot is running over 100 parachains, the projected TPS is well over 100,000. With [asynchronous backing upgrade](../learn/learn-async-backing), the TPS is expected to increase tenfold to 1,000,000.
```

CrackTheCode016 replied:
Love this - thank you so much Radha!

DrW3RK reviewed (commented):
Here is how I would answer this particular question. Feel free to edit or rephrase!

CrackTheCode016 committed:
Update docs/general/faq.md

Co-authored-by: Radha <86818441+DrW3RK@users.noreply.github.com>

CrackTheCode016 committed:
add stps

CrackTheCode016 committed:
polish

CrackTheCode016 commented:
@filippoweb3 @DrW3RK Thank you for the review. I added Radha's explaination, and added resources related to measuring TPS on Polkadot (which basically goes more in depth on the methodology, etc).

Commit:
polished text

- removed redundant parts, generalized

filippoweb3 reviewed (approved):
lgtm, I polished the text and removed things said twice
